src/helpers/apify.js

```diff
@@ -73,41 +72,53 @@ export function startCrawler(crawlerUrl, postData) {
       return response.json();
     })
     .then(function(data) {
-      apifyExecution = data;
-      keepPolling();
+      return keepPolling(data);
     });
-  return false;
 }
 
-export function keepPolling() {
-  console.log(apifyExecution);
-  if (apifyExecution.status !== 'SUCCEEDED' && apifyExecution.finishedAt === null) {
-    console.log('setting timeout');
-    setTimeout(function(){
-      console.log('calling ' + apifyExecution.detailsUrl);
-      fetch(apifyExecution.detailsUrl, {method: 'GET'})
+export function keepPolling(apifyExecution) {
+  return new Promise((resolve, reject) => {
+    console.log(apifyExecution);
+    if (apifyExecution.status !== 'SUCCEEDED' && apifyExecution.finishedAt === null) {
+      console.log('setting timeout');
+      setTimeout(function(){
+        console.log('calling ' + apifyExecution.detailsUrl);
+        fetch(apifyExecution.detailsUrl, {method: 'GET'})
```

Comment:
fetch was really just a hack when we didn't have promisified requests. In future we will want to change this I think. Not today.